### ISSUE TITLE
QNAP Container Station

### ISSUE THREAD

**Issue by wojowojo16** [jlesage/docker-crashplan-pro#325]
Forgive me, new to containers.....but I think I have this right.

If I deploy with GUI, I can remove /config "bind mount", add a new mount that mounts /config to /Containers/Crashplan-Pro/Config

Now, if I deploy with docker run it works, UNTIL I restart the container.

when I restart it mounts /config to a file within the container and not the mount point above.

Is this an issue with container station and how it works?   Is there a way to get around this?

Thanks!

my config:

docker run -d \
	--name=crashplan-pro \
	-e USER_ID=1000 \
	-e GROUP_ID=1000 \
	-e PUID=0 \
	-e PGID=0 \
	-e TZ=America/Chicago \
	-e CRASHPLAN_SRV_MAX_MEM=1024M \
	-p 5800:5800 \
	-p 5900:5900 \
	-v /share/containerapps/_ContainerConfigs/CrashplanPro/config:/config:rw \
	-v /share/containerapps/_ContainerConfigs/CrashplanPro/storage:/storage:rw \
	-v /share/Multimedia:/mnt/multimedia:rw \
	jlesage/crashplan-pro

**Comment by jlesage** [comment]
So when using the UI, can you create the same mapping as the cli?

**Comment by madfusker** [comment]
Hello,

Here is my config that has worked great on QNAP TVS-872XT for the last couple years.  FYI, the QNAP CS GUI is pretty much worthless...  Just use the docker run command below, and if you need to change anything, just kill the container and recreate it. I put my docker config files in a share called /share/Docker so they don't get destroyed if I kill the docker /and or upgrade it.  Everything else is ro on the shares unless I need to restore something.

Additionally, I noticed about a year or so ago that I had to start specifying the whole device path rather than /share/blah, otherwise the mappings would not work as they used to.  YMMV. 

docker run --restart unless-stopped -d \
	--name=crashplan-pro \
	--restart=always \
	-e USER_ID=0 \
	-e GROUP_ID=0 \
	-p 5800:5800 \
	-e TZ=America/Chicago \
	-e CRASHPLAN_SRV_MAX_MEM=16384M \
	-v /share/Docker/appdata/crashplan-pro:/config:rw \
	-v /share/CACHEDEV1_DATA/Docker:/share/Docker:ro \
	-v /share/CACHEDEV1_DATA/Logs:/share/Logs:ro \
	-v /share/CACHEDEV1_DATA/Public:/share/Public:ro \
	-v /share/CE_CACHEDEV2_DATA/Docs.Andrew:/share/Docs:ro \
	-v /share/CACHEDEV3_DATA/Movies:/share/Movies:ro \
	-v /share/CACHEDEV3_DATA/Music:/share/Music:ro \
	-v /share/CACHEDEV3_DATA/Pictures:/share/Pictures:ro \
	jlesage/crashplan-pro

MF

**Comment by wojowojo16** [comment]
@madfusker I did create with the docker command above, the problem is when the container was restarted it seemed to mount /config and /storage with the bind mount that is shown when deployed through the GUI, vs the specified mount to the /config file in my share.  

I wonder if I put the full path vs the share in if that will work?  I can try again and see what happens, I'd 100% rather use the docker run command.

**Comment by wojowojo16** [comment]
Still breaks on reboot.

docker run -d \
	--name=crashplan-pro \
	-e USER_ID=1000 \
	-e GROUP_ID=1000 \
	-e PUID=0 \
	-e PGID=0 \
	-e TZ=America/New_York \
	-e CRASHPLAN_SRV_MAX_MEM=32768M \
	-p 15800:5800 \
	-p 15900:5900 \
	-v /share/CACHEDEV3_DATA/containerapps/_ContainerConfigs/CrashplanPro/config:/config:rw \
	-v /share/CACHEDEV3_DATA/containerapps/_ContainerConfigs/CrashplanPro/storage:/storage:rw \
	-v /share/CE_CACHEDEV1_DATA/Multimedia:/mnt/multimedia:rw \
	-v /share/CE_CACHEDEV1_DATA/fvfc-qnap:/mnt/fvfc-qnap:rw \
	-v /share/CACHEDEV3_DATA/containerapps:/mnt/containerapps:rw \
	-v /share/CE_CACHEDEV1_DATA/all_data:/mnt/all_data:rw \
	jlesage/crashplan-pro

![image](https://user-images.githubusercontent.com/16326479/134554175-90635c51-d743-431c-9b06-3d301e989698.png)

**Comment by jlesage** [comment]
The host paths for `/config` and `/storage` don't look correct from the UI.

Instead of `-v /share/CACHEDEV3_DATA/containerapps/_ContainerConfigs/CrashplanPro/config:/config:rw`, could you use `-v /share/containerapps/_ContainerConfigs/CrashplanPro/config:/config:rw` ?

**Comment by madfusker** [comment]
Your configs are going to some strange location that I have never seen on any QNAP device before, and perhaps it's some temp storage location that gets wiped on reboot?

Simply create a share called "Docker" to store your docker files and then use "-v /share/Docker/appdata/crashplan-pro:/config:rw".  Problem solved and your settings will be there when you reboot!  :)

**Comment by wojowojo16** [comment]
@jlesage if you look at first post I used the -v /share/containerapps/_ContainerConfigs/CrashplanPr...and it works, but it doesn't remain persistent.  I reboot and get the drives setup like the picture I posted above.  I'm at a loss.

**Comment by madfusker** [comment]
> 
> Forgive me, new to containers.....
>
> ... the location is just mounted inside the container.
>
Right, which is why it gets destroyed.  It's doing exactly what you are telling it to do and writing to a tmp storage location.  If you don't know much about containers, follow the advice from others here that do and simply write the data to a persistent drive location.  This isn't an issue and can be closed.

**Comment by wojowojo16** [comment]
@madfusker can we start over?

docker run -d
--name=crashplan-pro
-e USER_ID=1000
-e GROUP_ID=1000
-e PUID=0
-e PGID=0
-e TZ=America/Chicago
-e CRASHPLAN_SRV_MAX_MEM=1024M
-p 5800:5800
-p 5900:5900
-v /share/containerapps/_ContainerConfigs/CrashplanPro/config:/config:rw
-v /share/containerapps/_ContainerConfigs/CrashplanPro/storage:/storage:rw
-v /share/Multimedia:/mnt/multimedia:rw
jlesage/crashplan-pro

Based on this, I have a share called "containerapps" which is an M2 SSD on my QNAP where I'm running container station and my containers.

Using the Docker run command above, I'm mounting /config and /storage to a folder name _ContainerConfigs/CrashplanPro/..., my config files are written to this location outside of the actual container.  The configuration works when I create the container.  The problem is if I STOP and START the container, the container itself mounts /config and /storage within the container, almost like I didn't create the mounts.  

If I deploy with the GUI, I can remove the default mounts and redirect.

![Capture_default](https://user-images.githubusercontent.com/16326479/135293098-bfc80f4a-2403-4516-88dd-383f9e70fb8f.JPG)
![Capture_newmounts](https://user-images.githubusercontent.com/16326479/135293127-ad11bfbe-4496-4437-8fbf-f1c9e42738b4.JPG)

**Comment by wojowojo16** [comment]
@jlesage @madfusker well Gents sorry to keep bugging you but I figured out what I was doing.

Deploying with Docker Run worked fine and was persistent.  After deploying I was going into container station, editing the container to limit the amount of memory / cpu used and setting it to auto-restart.  When I hit save and it restarted the container, it remounted /config and /storage inside the container folder, NOT in the folder I has assigned in my docker run command.  

Take-away is, deploy container and leave it alone and its fine.....

**Comment by madfusker** [comment]
Ah, yeah, that makes sense now...  We probably need a sticky side note in the instructions specifically for QNAP'ery, that you never want to change settings with the QNAP CS GUI.  I have never had anything work right using the GUI and never use it.  If I need to make any changes to the container I simply kill it and reissue the whole docker command on the command line.  Even to change memory or CPU settings.  Just delete the whole docker reissue the docker run command above.  Everything stays in place since the docker config/data files are outside the container.

**Comment by wojowojo16** [comment]
@madfusker yup that makes sense.  I'm an IT guy (heavy in infra/vmware etc), so I was 99% sure I was issuing command properly and was getting pretty pissed off when /config was getting remounted.    lol

**Comment by LeeTaylorX12** [comment]
Just as a PS to this exchange .. if you aren't used to the command line of docker I would recommend using Portainer to create the environment.  So much easier and quicker to find your way through.
